extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3Utils.java

```diff
@@ -242,4 +246,23 @@ public static S3ObjectSummary getSingleObjectSummary(ServerSideEncryptingAmazonS
 
     return objectSummary;
   }
+
+  /**
+   * Uploads a file to S3 if possible. First trying to set ACL to give the bucket owner full control of the file before uploading.
+   *
+   * @param service S3 client
+   * @param disableAcl true if ACL shouldn't be set for the file
+   * @param key The key under which to store the new object.
+   * @param file The path of the file to upload to Amazon S3.
+   */
+  public static void uploadFileIfPossible(ServerSideEncryptingAmazonS3 service, boolean disableAcl, String bucket, String key, File file)
+  {
+    final PutObjectRequest logFilePutRequest = new PutObjectRequest(bucket, key, file);
```

Comment:
This isn't strictly just for log files anymore, so how about naming this something more generic - even just `putObjectRequest`?

Reply:
yeah makes sense, will change that.

Reply:
Fixed.